pkg/cli/admin/release/extract_tools.go

```diff
@@ -199,36 +199,40 @@ func (o *ExtractOptions) extractCommand(command string) error {
 		currentOS = "darwin"
 	}
 
-	// select the subset of targets based on command line input
+	// Select the subset of targets based on command line input
 	var willArchive bool
 	var targets []extractTarget
+
+	// Filter by command, or gather all non-optional targets
 	if len(command) > 0 {
-		hasCommand := false
 		for _, target := range availableTargets {
-			if target.Command != command {
-				continue
-			}
-			hasCommand = true
-			if target.OS == currentOS || currentOS == "*" {
-				targets = []extractTarget{target}
-				break
+			if target.Command == command {
+				targets = append(targets, target)
 			}
 		}
-		if len(targets) == 0 {
-			if hasCommand {
-				return fmt.Errorf("command %q does not support the operating system %q", o.Command, currentOS)
-			}
-			return fmt.Errorf("the supported commands are 'oc' and 'openshift-install'")
-		}
 	} else {
-		willArchive = true
-		targets = availableTargets
+		for _, target := range availableTargets {
+			targets = availableTargets
+		}
+	}
+
+	// If the user didn't specify a command, or the operating system is set
+	// to '*', we'll produce an archive
+	if len(command) == 0 || o.CommandOperatingSystem == "*" {
 		for i := range targets {
 			targets[i].AsArchive = true
 			targets[i].AsZip = targets[i].OS == "windows"
 		}
 	}
 
+	if len(targets) == 0 {
+		if len(command) > 0 && currentOS != "*" {
+			return fmt.Errorf("command %q does not support the operating system %q", o.Command, currentOS)
+		} else {
+			return fmt.Errorf("the supported commands are 'oc', and 'openshift-install'")
```

Comment:
This error message is inaccurate now (there is no guarantee command has been specified).  You should also exit early, and the message should retain its original form.

```
switch {
case len(command) > 0 && currentOS != "*":
case len(command) > 0:
  return fmt.Errorf("the supported commands are 'oc' and 'openshift-install'")
default:
  return fmt.Errorf("no available commands")
}
```

to replace this if body.

Reply:
Ah good point. Updated.

Although, I assume the first case in your example above shouldn't fall through but return the `command %q does not support...` error?

Reply:
yes, sorry, I meant to add a `# add correct error here`